File: core-exercises/exercise-12/broken_calculator.c
#include <stdio.h

// Missing > in include above
// Missing function return types
// Missing semicolons
// Missing closing braces / parens

add(int a, int b)
    return a + b;
}

subtract(int a, int b) {
    return a - b
}

multiply(int a, int b {
    return a * b;
}

divide(int a, int b)
    if (b == 0) {
        printf("Error: Division by zero\n");
        return 0;
    }
    return a / b;
}

int main() {
    int x = 10, y = 5;

    printf("Addition: %d\n", add(x, y));
    printf("Subtraction: %d\n", subtract(x, y));
    printf("Multiplication: %d\n", multiply(x, y));
    printf("Division: %d\n", divide(x, y));

    return 0;

File: artifacts/core-exercises/exercise-12/broken_calculator.py
# Broken Python Calculator
# Intentional errors for Exercise 12

def add(a, b)
    return a + b  # Missing colon on def line above

def subtract(a, b)
    return a - b

def multiply(a, b)
    return a * b

def divide(a, b)
    if b = 0:  # Should be == not =
        print("Error: Division by zero")
        return None
    return a / b

def main(
    x = 10
    y = 5
    # Missing closing parenthesis on def main( above

    print(f"Addition: {add(x, y)}")
    print(f"Subtraction: {subtract(x, y)}")
    print(f"Multiplication: {multiply(x, y)}")
    print(f"Division: {divide(x, y)}")

main()
